public/apps/configuration/panels/audit-logging/audit-logging-edit-settings.tsx

```diff
@@ -0,0 +1,259 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiGlobalToastList,
+  EuiPageHeader,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { Toast } from '@elastic/eui/src/components/toast/global_toast_list';
+import { cloneDeep, pull, set } from 'lodash';
+import { AppDependencies } from '../../../types';
+import { SETTING_GROUPS, SettingContent } from './constants';
+import { EditSettingGroup } from './edit-setting-group';
+import { AuditLoggingSettings } from './types';
+import { buildHashUrl } from '../../utils/url-builder';
+import { ResourceType } from '../../types';
+import { updateAuditLogging } from '../../utils/audit-logging-view-utils';
+import { API_ENDPOINT_AUDITLOGGING } from '../../constants';
+
+interface AuditLoggingEditSettingProps extends AppDependencies {
+  setting: 'general' | 'compliance';
+}
+
+export function AuditLoggingEditSettings(props: AuditLoggingEditSettingProps) {
+  const [editConfig, setEditConfig] = useState<AuditLoggingSettings>({});
+  const [toasts, setToasts] = useState<Toast[]>([]);
+  const [invalidSettings, setInvalidSettings] = useState<string[]>([]);
+
+  const handleChange = (setting: SettingContent, val: boolean | string[]) => {
+    setEditConfig((previousEditedConfig) => {
+      return set(cloneDeep(editConfig), setting.path, val);
+    });
+  };
+
+  const handleInvalid = (key: string, error: boolean) => {
+    const invalid = pull([...invalidSettings], key);
+
+    if (error) {
+      invalid.push(key);
+    }
+    setInvalidSettings(invalid);
+  };
```

Comment:
Wondering that these 2 functions are similar, but are different at first argument. I would suggest use key for both,

Reply:
They are actually different. handleChange is for updating the configuration. handleInvalid is for error checking, specially this is only for the code editor.

Reply:
What I mean is unified to 
~~~
const handleChange = (key: string, val: boolean | string[]);
const handleInvalid = (key: string, error: boolean)
~~~
since handleChange only need key (path) field from setting.

Reply:
I see. Good idea. Fixed.